### PR TITLE
refactor: swap satori's uuid package to gofrs

### PR DESCRIPTION
**Pull request by mdanzinger**
**Reason for Change**:
Satori's go.uuid is effectively unmaintained and contained at least one critical security flaw which took half a year to be merged as the owner isn't active. See [satori/go.uuid#75](https://github.com/satori/go.uuid/pull/75)

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by tariq1890** [comment]
This is great!! Thank you @mdanzinger

**Comment by mboersma** [comment]
Some import errors were caught by `make test-style`.

**Comment by tariq1890** [comment]
@ultimateboy Are you okay with this Matt? IMO we should take this change as soon as possible. If you vendor aks-engine after this is merged, similar changes will have to be made in the projects that dep on aks-engine.

**Comment by tariq1890** [comment]
@amanohar @sozercan FYI. This should go into openshift as well.

File: cmd/root.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/Azure/aks-engine/pkg/helpers"
 	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/pkg/errors"
-	uuid "github.com/satori/go.uuid"
+	"github.com/gofrs/uuid"
```
Comment (tariq1890) [inline cmd/root.go]:
```suggestion
	uuid "github.com/gofrs/uuid"
```

File: cmd/root_test.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ package cmd
 import (
 	"testing"
 
-	uuid "github.com/satori/go.uuid"
+	"github.com/gofrs/uuid"
```
Comment (tariq1890) [inline cmd/root_test.go]:
```suggestion
	uuid "github.com/gofrs/uuid"
```

File: pkg/api/mocks.go [hunk 3]
```diff
@@ -5,13 +5,13 @@ package api
 
 import (
 	"github.com/Azure/go-autorest/autorest/to"
-	uuid "github.com/satori/go.uuid"
+	"github.com/gofrs/uuid"
```
Comment (tariq1890) [inline pkg/api/mocks.go]:
```suggestion
	uuid "github.com/gofrs/uuid"
```

File: pkg/armhelpers/graph.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/date"
 	"github.com/Azure/go-autorest/autorest/to"
-	uuid "github.com/satori/go.uuid"
+	"github.com/gofrs/uuid"
```
Comment (tariq1890) [inline pkg/armhelpers/graph.go]:
```suggestion
	uuid "github.com/gofrs/uuid"
```

**Comment by mdanzinger** [comment]
@mboersma  The only new errors I receive are :

``` cmd/root.go:17: File is not `gofmt`-ed with `-s` (gofmt)
	"github.com/pkg/errors"
cmd/deploy_test.go:15: File is not `gofmt`-ed with `-s` (gofmt)
	"github.com/pkg/errors"
pkg/api/vlabs/validate.go:20: File is not `gofmt`-ed with `-s` (gofmt)
	"github.com/pkg/errors"```

Is that what you're referring to?

File: Gopkg.lock [hunk 5]
```diff
@@ -2,15 +2,12 @@
 
 
 [[projects]]
-  digest = "1:f323f98930459f65f4699b5bfba563743680e2633d96e72d61a9732648e2d07c"
   name = "contrib.go.opencensus.io/exporter/ocagent"
```
Comment (tariq1890) [inline Gopkg.lock]: These changes should not happen. Please use `dep 0.5.0` when running `dep ensure`

**Comment by tariq1890** [comment]
@mdanzinger Can you make sure you `go install` the goimports installed in `go 1.11.5` and then run `goimports` to address the goimports issues?

File: Gopkg.lock (hunk 5 above)
Comment (mdanzinger) [reply Gopkg.lock]: I was using an outdating version of dep. I've updated it and pushed a commit.

**Comment by tariq1890** [comment]
@mdanzinger The codecov patch seems to fail. Could you rebase this PR?

**Comment by mdanzinger** [comment]
@tariq1890 I've rebased, squashed my previous commits, and pushed (I had to force push due to me accidentally merging two diverging branches), but it seems like the codecov patch is still failing, any ideas why?

**Comment by tariq1890** [comment]
@mdanzinger The codecov patch diff is failing because you have changed a line which has had no coverage previously. Normally, we'd request the contributor to add tests to fill that gap. In this case, adding test cases for the said method is less straightforward and would involve too much work which would be well out of the bounds of this PR's scope.

So this PR LGTM. I would defer to the other maintainers on assessing the impact of this dep change as some projects vendor in aks-engine as a dependency.

**Comment by tariq1890** [comment]
@jackfrancis @mboersma I have tested vendoring this fork against other repos that utilize aks-engine and I've found that it works with no issues.

**Comment by tariq1890** [comment]
Thanks @mdanzinger :)

**Pull request merged**